tianshou/policy/modelfree/dqn.py

```diff
@@ -128,15 +130,24 @@ def forward(self, batch: Batch,
         """
         model = getattr(self, model)
         obs = getattr(batch, input)
-        q, h = model(obs, state=state, info=batch.info)
+        obs_ = obs.obs if hasattr(obs, 'mask') else obs
+        q, h = model(obs_, state=state, info=batch.info)
         act = to_numpy(q.max(dim=1)[1])
+        if hasattr(obs, 'mask'):
```

Comment:
maybe it is better to use a flag. ``if hasattr(obs, 'mask'):`` occurs several times.